src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -121,9 +141,61 @@ bool BulletRigidObject::initializationFinalize(
 }  // initializeObjectFinalize
 
 void BulletRigidObject::finalizeObject() {
-  if (isUsingBBCollisionShape()) {
+  if (usingBBCollisionShape_) {
     setCollisionFromBB();
   }
+}  // finalizeObject
+
+std::unique_ptr<btCollisionShape>
+BulletRigidObject::buildPrimitiveCollisionObject(
+    assets::AbstractPrimitiveAttributes::ptr primAttributes) {
+  int primTypeVal = primAttributes->getPrimObjType();
+  assets::PrimObjTypes primType =
+      static_cast<assets::PrimObjTypes>(primTypeVal);
+  switch (primType) {
+    case assets::PrimObjTypes::CAPSULE_SOLID:
+    case assets::PrimObjTypes::CAPSULE_WF: {
+      // use bullet capsule :  btCapsuleShape(btScalar radius,btScalar height);
+      // auto attr =
+      // std::static_pointer_cast<assets::CapsulePrimitiveAttributes::ptr>(primAttributes);
+      btScalar radius = 1.0f;
+      btScalar height = 2.0 * primAttributes->getHalfLength();
+      return std::make_unique<btCapsuleShape>(radius, height);
+    }
+    case assets::PrimObjTypes::CONE_SOLID:
+    case assets::PrimObjTypes::CONE_WF: {
+      // use bullet cone : btConeShape(btScalar radius,btScalar height);
+      btScalar radius = 1.0f;
+      btScalar height = 2.0 * primAttributes->getHalfLength();
+      return std::make_unique<btConeShape>(radius, height);
+    }
+    case assets::PrimObjTypes::CUBE_SOLID:
+    case assets::PrimObjTypes::CUBE_WF: {
+      // use bullet box shape : btBoxShape( const btVector3& boxHalfExtents);
+      btVector3 dim(1.0, 1.0, 1.0);
+      return std::make_unique<btBoxShape>(dim);
+    }
+    case assets::PrimObjTypes::CYLINDER_SOLID:
+    case assets::PrimObjTypes::CYLINDER_WF: {
+      // use bullet cylinder shape :btCylinderShape (const btVector3&
+      // halfExtents);
+      btVector3 dim(1.0, 1.0, 1.0);
+      return std::make_unique<btCylinderShape>(dim);
+    }
+    case assets::PrimObjTypes::ICOSPHERE_SOLID:
+    case assets::PrimObjTypes::ICOSPHERE_WF:
+    case assets::PrimObjTypes::UVSPHERE_SOLID:
+    case assets::PrimObjTypes::UVSPHERE_WF: {
+      // use bullet sphere shape :btSphereShape (btScalar radius)
+      btScalar radius = 1.0f;
+      return std::make_unique<btSphereShape>(radius);
+    }
+    default: {
+      LOG(ERROR) << "Unknown PrimObjTypes value : " << primTypeVal
+                 << " | Aborting.";
```

Comment:
This failure isn't happening much in practice, right? Only when there's a new primitive and this code didn't get updated accordingly. I would assert here.